mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/BaseRoute.java

```diff
@@ -322,7 +323,12 @@ Boolean parseBoolean(String val) {
     protected  <T> Route withFuture(CompletableFuture<T> tFuture) {
         return onComplete(tFuture,
             t -> t.fold(
-                throwable -> complete(StatusCodes.INTERNAL_SERVER_ERROR, throwable, Jackson.marshaller()),
+                throwable -> {
+                    if (throwable instanceof ResourceNotFoundException) {
```

Comment:
I would move ResourceNotFoundException to some proto package (that's user-facing) to imply that this is part of the API and then have TaskExecutorNotFoundException extend it.